### PR TITLE
Add pytorch reference implementation of cosmoflow

### PR DESCRIPTION
**Pull request by fiedorowicz1**
Adds a pytorch reference implementation of cosmoflow utilizing distributed data parallel and automatic mixed precision.

### CODE REVIEW

File: applications/physics/cosmology/cosmoflow/ref/batch.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+#BSUB -J cosmoflow
+#BSUB -q pbatch
+#BSUB -alloc_flags ipisolate
+
+firsthost=`jsrun --nrs 1 -r 1 /bin/hostname`
+export MASTER_ADDR=$firsthost
+export MASTER_PORT=23456
+
+echo "Started at $(date)"
+jsrun -r 4 --bind none -c 10 python main.py --preload --enable-amp --use-batchnorm
```
Comment (ndryden) [inline applications/physics/cosmology/cosmoflow/ref/batch.sh:16]: Minor thing, but I recall that jsrun/SMPI can sometimes complain if you never initialize MPI (which we don't, due to using NCCL). Passing `--smpiargs="off"` avoids this.

File: applications/physics/cosmology/cosmoflow/ref/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# PyTorch Reference Implementation of CosmoFlow
+
+This directory contains a reference implementation of CosmoFlow in PyTorch as it is implemented for LBANN in the parent directory. This implementation uses `DistributedDataParallel` (DDP) to run on multiple GPUs. The included `batch.sh` script provides an example for training this model with DDP on Lassen.
```
Comment (ndryden) [inline applications/physics/cosmology/cosmoflow/ref/README.md:3]: You might also note that the script hardcodes OpenMPI env variables, so even if you write a new batch script, it still may not run on other systems.

**Review by ndryden — APPROVED** [review]
Looks reasonable to me. Just a couple suggestions to clarify things.

File: applications/physics/cosmology/cosmoflow/ref/README.md (hunk 2 above)
Comment (fiedorowicz1) [reply applications/physics/cosmology/cosmoflow/ref/README.md]: Fixed in https://github.com/LLNL/lbann/pull/2199/commits/05265916be3ce2adafa5bf68da350d6e3d680b14

**Pull request merged**